src/Database/LSMTree/Internal/Merge.hs

```diff
@@ -131,11 +181,22 @@ complete ::
   => Merge m h
   -> m (Run m (FS.Handle h))
 complete Merge{..} = do
-    readMutVar mergeLastStepResult >>= \case
-      MergeInProgress -> error "complete: Merge is not yet completed!"
-      MergeComplete -> do
-        Run.fromMutable mergeHasFS mergeHasBlockIO mergeCaching
-                        (RefCount 1) mergeBuilder
+    readMutVar mergeState >>= \case
```

Comment:
Right now, no, since `Run.fromMutable` calls `RunBuilder.unsafeFinalise` which is not safe to do twice. We would have to rework the run builder a little bit

Reply:
Okay, then maybe we should put a TODO stating this in relevant section of code.